File: UPTMDigital.API/Models/Constancia.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace UPTMDigital.API.Models
{
    [Table("Constancia")]
    public class Constancia
    {
        [Key]
        public int IdConstancia { get; set; }
        public int EstudianteId { get; set; }
        public string TipoConstancia { get; set; } = null!;
        public string? Estado { get; set; }
        public DateTime? FechaSolicitud { get; set; }
        public string? CodigoQR { get; set; }
        public string? ArchivoUrl { get; set; }

        public Estudiante Estudiante { get; set; } = null!;
    }
}

File: uptmdigital_app/lib/models/constancia.dart
class Constancia {
  final int idConstancia;
  final int estudianteId;
  final String tipoConstancia;
  final String estado;
  final String fechaSolicitud;
  final String codigoQR;

  Constancia({
    required this.idConstancia,
    required this.estudianteId,
    required this.tipoConstancia,
    required this.estado,
    required this.fechaSolicitud,
    required this.codigoQR,
  });

  factory Constancia.fromJson(Map<String, dynamic> json) {
    return Constancia(
      idConstancia: json['idConstancia'] ?? 0,
      estudianteId: json['estudianteId'] ?? 0,
      tipoConstancia: json['tipoConstancia'] ?? '',
      estado: json['estado'] ?? '',
      fechaSolicitud: json['fechaSolicitud'] != null ? json['fechaSolicitud'].toString() : '',
      codigoQR: json['codigoQR'] ?? '',
    );
  }
}
